### ISSUE TITLE
Question: Implementable with multiple codepoint glyphs?

### ISSUE THREAD

**Issue by TJSomething** [qntm/base131072#6]
Per [Bear plus snowflake equals polar bear](https://andysalerno.com/posts/weird-emojis/), you can store extra data per tweet with multiple codepoint glyphs. Are there enough such glyphs to implement this?

**Comment by qntm** [comment]
Base131072 is (or will be) optimised for UTF-32-encoded text, not Twitter. Polar bear is U+1F43B U+200D U+2744 U+FE0F which in UTF-32 is a fixed 16 bytes - there are no savings to be had here.

For encoding data in Twitter, our best effort encoding right now is [Base2048](https://github.com/qntm/base2048), which encodes 3,080 bits = 385 bytes per 280-"character" Tweet, or 11 bits per "character".

The essay states,

> regardless of the byte count of a Unicode emoji character, it will never count as more than 2 characters

In order to outperform Base2048, we need to encode more than 11 bits per character, which is 22 bits per 2 characters, which is 22 bits per emoji. In order words we need a repertoire of more than 2^22 = 4,194,304 distinct emojis to play with. To make a reasonable improvement, let's say encoding 23 bits per emoji (3,220 bits = 402.5 bytes per 280-"character" Tweet), we actually need *twice* that many emojis: 8,388,608.

So in short the answer is no! :) Thanks for a great question.
